noxfile.py

```diff
@@ -0,0 +1,56 @@
+"""Experimental nox configuration file."""
+
+import nox
+
+nox.options.default_venv_backend = "uv"
+
+
+maxpython = "3.12"
+
+doc_requirements = ("-r", "requirements/requirements_docs_py312.txt")
+
+sphinx_commands = (
+    "sphinx-build",
+    "docs/",
+    "docs/build/html",
+    "--nitpicky",
+    "--fail-on-warning",
+    "--keep-going",
+)
+
+html = ("-b", "html")
+check_hyperlinks = ("-b", "linkcheck")
```

Comment:
📝 By defining the options into discrete groups like this, we can compose different environments to build the docs 📚 with different option sets ⚙️, which should make the commands more reusable. I suspect it'll be more maintainable too, since the commands that get reused are isolated from the details of the testing environment ↔️.